contracts/PixelmonPartySquad/PxPartySquad.sol

```diff
@@ -0,0 +1,391 @@
+            IERC721 erc721Contract = IERC721(_treasure.collectionAddress);
+            if (_treasure.tokenIds.length < _treasure.claimedToken) {
+                revert InsufficientToken();
+            }
+            erc721Contract.transferFrom(vaultWalletAddress, msg.sender, _treasure.tokenIds[_treasure.claimedToken - 1]);
+            emit TreasureTransferred(_weekNumber, msg.sender, _treasure.collectionAddress,_treasure.tokenIds[_treasure.claimedToken - 1] , _treasure.contractType);
+        }
+
+        
+    }
+
+    /// @notice Set treasure distributions for a week
+    /// @dev Only admin can call this method
+    /// @param _weekNumber The week number
+    /// @param _treasureindexes The index of the treasure in 'treasures' mapping variable
+    /// @param _treasureCounts Amount of treasure that will be available to claim during the week
+    /// @param _specialTreasuresCount Amount of special treasure that will be available to claim during the week
+    function setWeeklyTreasureDistribution(
+        uint256 _weekNumber,
+        uint8[] memory _treasureindexes,
+        uint16[] memory _treasureCounts,
+        uint8 _specialTreasuresCount
+    ) external onlyAdmin(msg.sender) validTreaureDistributionPeriod(_weekNumber) validArrayLength(_treasureindexes.length, _treasureCounts.length) {
+        Week storage week = weekInfos[_weekNumber];
+        week.specialTreasureCount = _specialTreasuresCount;
+        week.availableSpecialTreasureCount = _specialTreasuresCount;
+        week.treasureCount = 0;
+        for (uint256 index = 0; index < _treasureindexes.length; index = _uncheckedInc(index)) {
+            if (_treasureindexes[index] == 0 || _treasureindexes[index] > totalTreasureCount) {
+                revert InvalidTreasureIndex();
+            }
+            week.treasureCount++;
+            week.distributions[week.treasureCount].treasureIndex = _treasureindexes[index];
+            week.distributions[week.treasureCount].totalSupply = _treasureCounts[index];
+            week.remainingSupply += _treasureCounts[index];
+        }
+    }
+
+    /// @notice Set a list of winners for a particular week
+    /// @param _weekNumber The current week number
+    /// @param _winners List of wallet addresses that have been selected as winners
+    /// @param _treasureCounts Amount of treasure that have been awarded to the corresponding winner
+    function updateWeeklyWinners(
+        uint256 _weekNumber,
+        address[] memory _winners,
+        uint8[] memory _treasureCounts
+    ) external onlyModerator(msg.sender) validArrayLength(_winners.length, _treasureCounts.length) validWinnerUpdationPeriod(_weekNumber) {
+        for (uint256 index = 0; index < weekInfos[_weekNumber].specialTreasureWinners.length; index++) {
+            address specialTreasureWinner = weekInfos[_weekNumber].specialTreasureWinners[index];
+            weekInfos[_weekNumber].specialTreasureWinnerMap[specialTreasureWinner] = false;
```

Comment:
specialTreasureWinnerMap should be bool map or a count map?

Reply:
boolean as for each week, one person can win max one special treasure